sccameraview/src/main/java/com/example/sccameraview/SaveImageTask.java

```diff
@@ -0,0 +1,64 @@
+package com.example.sccameraview;
+
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import static com.example.sccameraview.BaseCameraView.LOG_TAG;
+
+class SaveImageTask extends AsyncTask {
+
+    OnImageSavedListener imageSavedListener;
+    private byte[] imageData;
+    private File imageFile;
+
+    SaveImageTask(OnImageSavedListener imageSavedListener) {
+        this.imageSavedListener = imageSavedListener;
+    }
+
+    @Override
+    protected Object doInBackground(Object[] objects) {
+        FileOutputStream fos = null;
+        try {
+            if (imageFile == null){
+                Log.d(LOG_TAG, "Error creating media file, check storage permissions");
+                return null;
+            }
+            if (imageData == null){
+                Log.d(LOG_TAG, "No image data");
+                return null;
+            }
+            try {
+                //TODO: test rotation and orientation on multiple devices
+                fos = new FileOutputStream(imageFile);
+                fos.write(imageData);
+                fos.close();
+            } catch (FileNotFoundException e) {
+                Log.d(LOG_TAG, "File not found: " + e.getMessage());
+            } catch (IOException e) {
+                Log.d(LOG_TAG, "Error accessing file: " + e.getMessage());
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return fos;
+    }
+
+    @Override
+    protected void onPostExecute(Object o) {
+        imageSavedListener.onImageSaved();
```

Comment:
I am confused. We introduced this listener so the user can set arbitrary action after the image is taken. But now i do not even see the setter for that listener.